From 06ea6b381117a4f49c2854bb54efb099b7295276 Mon Sep 17 00:00:00 2001
From: Robert Carr <racarr@google.com>
Date: Tue, 22 Mar 2022 15:25:54 -0700
Subject: [PATCH] DO NOT MERGE: SurfaceControl: Ensure unused fields are zeroed

Otherwise we could leak uninitialized memory to
an untrusted process.

Bug: 214999987
Test: Existing tests pass
Change-Id: I87993fbb920b3af938a311e6afc15383d571823a
(cherry picked from commit c2cf02c17af0d7e5f58b8a17ecb9cfcc91a4d2eb)
(cherry picked from commit 34fe38b0c7ce6a47eaa2f328865858a2c7dac5e1)
Merged-In: I87993fbb920b3af938a311e6afc15383d571823a
---
 libs/gui/SurfaceControl.cpp           |  1 +
 libs/gui/include/gui/SurfaceControl.h | 12 ++++++------
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/libs/gui/SurfaceControl.cpp b/libs/gui/SurfaceControl.cpp
index 6529a4e51..f93b79975 100644
--- a/libs/gui/SurfaceControl.cpp
+++ b/libs/gui/SurfaceControl.cpp
@@ -70,6 +70,7 @@ SurfaceControl::SurfaceControl(const sp<SurfaceControl>& other) {
     mLayerId = other->mLayerId;
     mWidth = other->mWidth;
     mHeight = other->mHeight;
+    mFormat = other->mFormat;
     mCreateFlags = other->mCreateFlags;
 }
 
diff --git a/libs/gui/include/gui/SurfaceControl.h b/libs/gui/include/gui/SurfaceControl.h
index 9ee4636ae..9a423f245 100644
--- a/libs/gui/include/gui/SurfaceControl.h
+++ b/libs/gui/include/gui/SurfaceControl.h
@@ -118,12 +118,12 @@ private:
     mutable sp<Surface>         mSurfaceData;
     mutable sp<BLASTBufferQueue> mBbq;
     mutable sp<SurfaceControl> mBbqChild;
-    int32_t mLayerId;
-    uint32_t mTransformHint;
-    uint32_t mWidth;
-    uint32_t mHeight;
-    PixelFormat mFormat;
-    uint32_t mCreateFlags;
+    int32_t mLayerId = 0;
+    uint32_t mTransformHint = 0;
+    uint32_t mWidth = 0;
+    uint32_t mHeight = 0;
+    PixelFormat mFormat = PIXEL_FORMAT_NONE;
+    uint32_t mCreateFlags = 0;
 };
 
 }; // namespace android
-- 
2.37.0.rc0.161.g10f37bed90-goog

